### ISSUE TITLE
Modifier l'intitulé du commit de merge est-il encore nécessaire ? 

### ISSUE THREAD

**Issue by sandcha** [openfisca/openfisca-france#1563]
Hello hello !

Je suis le fan numéro un d'OpenFisca, mais je rencontre un p'tit problème au quotidien.

### Qu'ai-je fait ?

Depuis un moment (2017 ?), on encourage les contributeurs qui mergent une PR à mettre pour nom de commit le nom de la PR afin que le sens de chaque commit de merge sur `master` soit indiqué d’un coup d’oeil. 

### À quoi m'attendais-je ?

À que cette opération mineure se fasse presque automatiquement pour nous tous.

### Que s'est-il passé en réalité ?

Nous n'avons pas écrit cette règle : ni dans le [CONTRIBUTING.md](https://github.com/openfisca/openfisca-france/blob/f494684caf719c81ac8ba420b272911ee634fcd7/CONTRIBUTING.md) ni sur le wiki (comme, par exemple, dans notre ébauche de [Guide de la collaboration sur GitHub](https://github.com/openfisca/openfisca-france/wiki/Guide-de-la-collaboration-sur-GitHub)).

Or, cette petite opération nécessite d’être attentif pile quand à notre grande joie, on peut cliquer sur le bouton vert de merge :tada: (en inversant le texte de commit proposé par défaut par GitHub avec le commentaire=nom de la PR 🔍 ).
Et c'est une des règles implicites qu’on transmet à chaque nouvel arrivant qui a déjà beaucoup de choses à intégrer. 

Il me semble que la “charge mentale” de cette petite règle dépasse l’apport qu’elle a maintenant que GitHub met un lien derrière chaque #ID-PR.

Par exemple : 
"Merge pull request #1443 from openfisca/bump-pytest" (message par défaut proposé par GitHub)
ou "Met à jour la dépendance pytest" (nom de la PR)
permettent toutes deux d'avoir une idée de ce qui a été intégré et le changelog est aussi là pour donner plus d'information.

Je pense donc que nous devrions soit supprimer cette règle (dont l'apport, me semble-t-il, est quand même positif quand on consulte l'historique dans un terminal), soit l'inscrire quelque part dans les bonnes pratiques de contribution pour qu'elle ne soit plus à porter oralement 🙂 

### Voici des informations qui peuvent aider à reproduire le problème :

L'[historique des commits](https://github.com/openfisca/openfisca-france/commits/master) sur france où on oublie régulièrement cette règle.

C'est aussi une pratique encouragée sur tous les autres dépôts de l'organisation openfisca.

### Contexte

Je m'identifie plus en tant que :

- [x] Contributeur·e : je contribue à OpenFisca France.
- [ ] Développeur·e : je crée des outils qui utilisent OpenFisca France.
- [ ] Économiste : je réalise des simulations avec des données.
- [x] Mainteneur·e : j'intègre les contributions à OpenFisca France.
- [ ] Autre : _(ajoutez une description du contexte dans lequel vous utilisez OpenFisca)_.

**Comment by MattiSG** [comment]
Merci @sandcha d'avoir pris le temps de documenter cette règle et de poser cette question !

Pour ma part, je suis convaincu de l'importance d'utiliser un nom complet plutôt que de se reposer sur le message généré automatiquement, pour les raisons suivantes :

1. Le lien automatique vers l'identifiant fait l'hypothèse que la consultation a lieu sur GitHub, et donc qu'un lien est bien généré ; sinon, il faut avoir connaissance de la signification de cet identifiant et aller le chercher manuellement sur GitHub.
2. Le lien automatique vers l'identifiant fait l'hypothèse que GitHub restera publiquement accessible pour une durée aussi longue que l'historique d'OpenFisca ; si on intègre l'information dans git, elle restera par définition accessible pour une durée aussi longue que l'historique d'OpenFisca.
3. Le lien automatique vers l'identifiant fait l'hypothèse qu'OpenFisca sera toujours hébergé sur GitHub ; si on migre sur GitLab par exemple, les liens seront cassés puisqu'ils feront référence à des identifiants inexistants voire pire, erronés.
4. La consultation de l'historique hors GitHub est grandement facilitée par l'usage d'un intitulé auto-contenu (qui ne se repose pas sur la consultation d'une ressource externe). En particulier, je pense aux opérations de type `rebase` et `bisect`, pour lesquelles la manipulation est vraiment plus simple lorsqu'on peut se reposer sur un titre de commit clair.
5. Dans OpenFisca-France, les messages de commit sont en français. Je trouve incohérent que certains soient en anglais.

Je suis donc en faveur de l'incorporation de cette règle dans le CONTRIBUTING.md 🙂

**Comment by benjello** [comment]
Je me rallierai à la majorité.

**Comment by sandcha** [comment]
Effectivement, si l'on sort un jour de GitHub (chose probable dans la très longue vie à venir d'openfisca ^^), les messages de commit explicites/auto-contenus me semblent aussi préférables.
Ok pour que ce soit dans le CONTRIBUTING.

Globalement, toute source écrite évitera ce sentiment d'ergoter sur des vétilles à chaque rappel fait de la règle. 🙂
